addons/mrp/models/mrp_workorder.py

```diff
@@ -103,21 +103,53 @@ class MrpWorkorder(models.Model):
     capacity = fields.Float(
         'Capacity', default=1.0,
         help="Number of pieces that can be produced in parallel.")
-    workorder_line_ids = fields.One2many('mrp.workorder.line', 'workorder_id', string='Workorder lines')
+    workorder_line_ids = fields.One2many(
+        'mrp.workorder.line',
+        'workorder_id',
+        string='Workorder lines',
+        domain=[('finished_check', '=', False)]
+    )
+    final_lot_line_ids = fields.One2many(
+        'mrp.workorder.line',
+        'workorder_id',
+        string='Final Lot lines',
+        domain=[('finished_check', '=', True)]
+    )
     final_lot_domain = fields.One2many('stock.production.lot', compute="_compute_final_lot_domain")
 
-    @api.depends('production_id')
+    @api.onchange('final_lot_id')
+    def _onchange_final_lot_id(self):
+        previous_wo = self.env['mrp.workorder'].search([
+            ('next_work_order_id', '=', self.id)
+        ])
+        # update default quantity to produce depending on the previous qty_producing
+        # in the previous workorder for the current final lot
+        if previous_wo:
+            line = previous_wo.final_lot_line_ids.filtered(lambda line: line.lot_id == self.final_lot_id)
+            if line:
+                self.qty_producing = line.qty_done
+
     def _compute_final_lot_domain(self):
-        for wo in self:
-            # check if self is not the first workorder in the list
-            if self.env['mrp.workorder'].search([('next_work_order_id', '=', wo.id)]):
-                wo.final_lot_domain = self.env['stock.production.lot'].search([
-                    ('use_next_on_work_order_id', '=', wo.id),
-                ]).ids
-            else:
+        """ browse already created lot and respective quantities"""
```

Comment:
not good performance wise
the compute will be the same for all workoders of a same production so you should assign them all at once
``` python
productions = self.mapped('production_id')
for production in productions:
    if production.product_id.tracking ==  'none':
        continue

    finished_workorder_lines = production.wokorder_ids.mapped('finished_workorder_line_ids')
    qty_dones_per_lot = defaultdict(list)
    for finished_workorder_line in finished_workorder_lines:
        qty_dones_per_lot[finished_workorder_line.lot_id.id].append(finished_workorder_line.qty_done)

    qty_to_produce = production.product_qty
    allowed_lot_ids = self.env['stock.production.lot'
    qty_produced = sum([max(qty_dones) for qty_dones in qty_dones_per_lot.values()])
    if float_compare(qty_produced, qty_to_produce, rounding=...) < 0:
        # If we haven't produced enough, all lots are available
        allowed_lot_ids = self.env['stock.production.lot'].search([('product_id', '=', production.product_id.id)])
    else:
        # If we produced enough, only the already produced lots are available
        allowed_lot_ids = self.env['stock.production.lot'].browse(qty_dones_per_lot.keys())
    production.workorder_ids.write({final_lot_domain: allowed_lot_ids})

# in the view add a compute produced_lot_ids on the workorder
'|',
    '&',
        ('product_tracking', '=', 'lot'),
        ('final_lot_id', 'in', 'final_lot_domain'),
    '&',
        ('product_tracking', '=', 'serial'),
        '&',
             ('final_lot_id', 'in', 'final_lot_domain'),
             ('final_lot_id', 'not in', 'produced_lot_ids'),

rename available by allowed
```